simple-service-webapp-rest/src/main/java/com/example/repository/TokenRepository.java

```diff
@@ -0,0 +1,45 @@
+package com.example.repository;
+
+
+import com.example.authentication.Token;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+public class TokenRepository {
+
+    private static List<Token> tokens = new ArrayList();
+
+    public static void addToken(Token newToken){
+        tokens.add(newToken);
+    }
+
+    public static List<Token> getTokens(){
+        return Collections.unmodifiableList(tokens);
+    }
+
+    public static Token containsValidToken(String username){
```

Comment:
Here i check if the user already has a valid token that is if the token is not expired. if he already has a  valid token i just give it back to, else ill issue a new one. 
Scenario: user logs in ->accidentally closes the tab -> user logs in again within a minute, the same token could be given back, no need for a new one.